lib/active_model/serializer/collection_reflection.rb

```diff
@@ -2,6 +2,12 @@ module ActiveModel
   class Serializer
     # @api private
     class CollectionReflection < Reflection
+      def initialize(*)
+        super
+        @class_name = options.fetch(:class_name, name.to_s.camelize.singularize)
+        @type = class_name.underscore.pluralize.to_sym
+        @foreign_key ||= "#{name.to_s.singularize}_ids".to_sym
```

Comment:
`class_name` and `type` should be used in more cases for inferring a serializer, rather that requiring an instantiated associated object.  This can move us in a direction where a serializer on its own an only serialize itself and its associations and only an adapter can get it to serialize deeper. And each association on the serializer can have the serializer set else is inferred.